File: src/main/java/com/himadri/model/UserRequest.java
package com.himadri.model;

import java.io.InputStream;

public class UserRequest {
    private final String requestId;
    private final InputStream csvInputStream;
    private final String catalogueTitle;
    private final boolean draftMode;

    public UserRequest(String requestId, InputStream csvInputStream, String catalogueTitle, boolean draftMode) {
        this.requestId = requestId;
        this.csvInputStream = csvInputStream;
        this.catalogueTitle = catalogueTitle;
        this.draftMode = draftMode;
    }

    public String getRequestId() {
        return requestId;
    }

    public InputStream getCsvInputStream() {
        return csvInputStream;
    }

    public String getCatalogueTitle() {
        return catalogueTitle;
    }

    public boolean isDraftMode() {
        return draftMode;
    }
}

File: src/main/java/com/himadri/dto/UserRequest.kt
package com.himadri.dto

import java.io.InputStream

class UserRequest(
    val requestId: String,
    val csvInputStream: InputStream,
    val catalogueTitle: String,
    val quality: Quality,
    val isWholeSaleFormat: Boolean,
    val isAutoLineBreakAfterMinQty: Boolean,
    val headerImageStream: InputStream?,
    val isWideHeaderImage: Boolean,
    val footerImageStream: InputStream?,
    val isWideFooterImage: Boolean
)
